sdk/go.mod

```diff
@@ -0,0 +1,22 @@
+module github.com/c3os-io/c3os/sdk
+
```

Comment:
small nit: two go.mod would likely collide, is there a specific reason for it?

Reply:
Go modules support nesting like this.  It differentiates the requirements of the sdk from c3os as a whole.  It reduces the number of collisions when someone is writing an extension for c3os.  We could also solve it by moving the sdk to another repo but I prefer the sdk is released with the core framework as they are tightly coupled.